packages/lodestar/src/api/impl/validator/validator.ts

```diff
@@ -119,6 +116,8 @@ export class ValidatorApi implements IValidatorApi {
 
   public async publishBlock(signedBlock: SignedBeaconBlock): Promise<void> {
     await this.checkSyncStatus();
+    await waitForBlockSlot(this.config, this.chain.getGenesisTime(), signedBlock.message.slot);
+    this.chain.clock.currentSlot;
```

Comment:
this line seems useless?

Reply:
I added this to sync the clock state with the current time. (so that forkchoice ERR_FUTURE_SLOT couldn't be triggered)
Calling `clock.currentSlot` is a getter, which can potentially update the current slot if the time has changed.
If we have a `waitForSlot` method on clock, or something functionally similar, then we won't have to worry about synchronization with the clock.